### ISSUE TITLE
Looks like a bug in dagbuilder.py

### ISSUE THREAD

**Issue by ilaichi** [ajbosco/dag-factory#101]
The condition on 197 looks like a bug. As both are required, there should be **or** instead of **and**. 
![image](https://user-images.githubusercontent.com/7289686/130313056-182fd822-30a2-43c1-8a10-f107461ed709.png)

**Comment by brucemen711** [comment]
Yes, I think we need a fix for this. Because python_callable_file and python_callable_name are required parameters to construct a Python Operator.

**Comment by tatiana** [comment]
Thanks for flagging this, and apologies for the very long silence on this ticket!

I took a look at the current code. The exact block from your screenshot no longer exists — `dagbuilder.py` has been refactored quite a bit since 2021 (the check now uses `issubclass(operator_obj, PYTHON_CALLABLE_CLASSES)`, also covers `PythonSensor`, and a third `python_callable` dotted-path option was added).

However, the underlying issue you pointed at does still seem to be present. The guard in `make_task` is now (pinned to `main` @ `7beb222`):

https://github.com/astronomer/dag-factory/blob/7beb22241b70ddd49f2fd2f61225570a266c8ed4/dagfactory/dagbuilder.py#L344-L356

Because the guard only raises when *all three* are missing, supplying just **one** of `python_callable_name` / `python_callable_file` (and no `python_callable`) still slips past validation. It then hits the direct `task_params["..."]` access just below:

https://github.com/astronomer/dag-factory/blob/7beb22241b70ddd49f2fd2f61225570a266c8ed4/dagfactory/dagbuilder.py#L357-L360

…which raises a `KeyError` instead of the intended, friendly `DagFactoryException`. So your original "`and` should arguably be `or`" observation is still relevant — just adapted to the new three-option structure.

We'll keep this open as a small validation bug. A fix would be to detect when exactly one of the pair is provided and raise a clear `DagFactoryException` explaining that `python_callable_name` and `python_callable_file` must be supplied together.

(Note: I traced this by reading the code rather than running it, so a quick reproducing test would be good to confirm the `KeyError` path.)

**Comment by cen-zp** [comment]
Hi! I’d like to work on this issue.

I plan to add explicit validation for partially supplied `python_callable_name` / `python_callable_file` pairs, update the existing broad exception test to assert `DagFactoryException`, and cover both missing-half cases. I’ll run the focused `test_dagbuilder.py` tests and pre-commit checks. Does this scope match your expectation?

**Comment by pankajastro** [comment]
Hi @cen-zp, I've assigned this issue to you.
